lab_1_classify_by_unigrams/main.py

```diff
@@ -6,55 +6,103 @@
 
 def tokenize(text: str) -> list[str] | None:
     """
-    Splits a text into tokens, converts the tokens into lowercase,
-    removes punctuation, digits and other symbols
-    :param text: a text
-    :return: a list of lower-cased tokens without punctuation
-    """
+        Splits a text into tokens, converts the tokens into lowercase,
+        removes punctuation, digits and other symbols
+        :param text: a text
+        :return: a list of lower-cased tokens without punctuation
+        """
+    if not isinstance(text, str):
+        return None
+    tokens = []
+    for symbol in text:
+        if symbol.isalpha():
+            tokens += symbol.lower()
+    return tokens
 
 
 def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
     """
-    Calculates frequencies of given tokens
-    :param tokens: a list of tokens
-    :return: a dictionary with frequencies
-    """
+        Calculates frequencies of given tokens
+        :param tokens: a list of tokens
+        :return: a dictionary with frequencies
+        """
+    if (not isinstance(tokens, list)
+            or not tokens
+            or not all(isinstance(element, str) for element in tokens)):
+        return None
+    frequencies = {}
+    unit = 1 / len(tokens)
+    for letter in tokens:
+        if letter in frequencies:
```

Comment:
What if we set 0 if `letter` is not in `frequencies` and always do += ...

Reply:
Thank you, implemented